### PR TITLE
JSON API and views

### LINKED ISSUE #6: Controllers JSON actions
Finish up Controllers to make it fully functional with JSON REST API

### PR DESCRIPTION
**Pull request by mszynka**
Adds JSON API in controllers
Adds ModelEditors, Create, Edit and Delete views
Adds Logic for actions
Reimplements contollers
Adds menu dropdowns
Closes #6

### CODE REVIEW

**Comment by mszynka** [comment]
I'm not so sure about those views. @bojakowsky Should I use ViewModel or just Model?

**Comment by bojakowsky** [comment]
ViewModel as name suggests is for the view, I've just started correcting stuff

**Comment by mszynka** [comment]
Yeah, I've made a mistake and fixed it. Merge #25 to this for feature and cleanup.

**Comment by bojakowsky** [comment]
I go to sleep, I'll end it up as I wake up. 
Closed #23

**Comment by bojakowsky** [comment]
- UserID in database is a string. That's why, i'll correct it
- var ass in assignments was funny
- BaseSerializer... it's not hardcoded, it's just how mapper MUST work. one type enum =! all other enums, but however I had no time to check if typeof(enum) will work
- Sorry for building eror, it was really later hours, and miss typed types, lol.

**Comment by mszynka** [comment]
I don't mind build error :wink: And I didn't read the commit message. Sorry :imp: 

Anyway this `typeof(enum)` would be nice. Or could we implement like a `enum interface` and then `typeof` would be explicit. We should be able to. It's a :cl: class, isn't it?

**Comment by bojakowsky** [comment]
If you're okay with the changes merge PR and close it.
I left many things unimplemented, I'll open another issue for that tomorrow.
Mainly i structurized code, made it more objective.
Contact me via facebook about the eventual problems, I'm having to much spam on gmail to handle that.
fixed #6 #30

File: TaskVlopper.Base/Logic/IMeetingLogic.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TaskVlopper.Base.Logic
+{
+    public interface IMeetingLogic
+    {
+        IEnumerable<TaskVlopper.Base.Model.Meeting> GetAllMeetingsForCurrentUser(string userId);
+        IEnumerable<TaskVlopper.Base.Model.Meeting> GetAllMeetingsForCurrentUserAndProject(string userId, int projectId);
+        IEnumerable<TaskVlopper.Base.Model.Meeting> GetAllMeetingsForCurrentUserAndProjectAndTask(string userId, int projectId, int taskId);
```
Comment (mszynka) [inline TaskVlopper.Base/Logic/IMeetingLogic.cs:14]: Why 3 methods for the same task? Couldn't we use like `GetAllMeetingsForCurrent(string userId = null, int projectId = null...)`? And then you would just use `GetAllMeetingsForCurrent(userId: mystring)`

File: TaskVlopper.Base/Logic/IWorklogLogic.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TaskVlopper.Base.Logic
+{
+    public interface IWorklogLogic
+    {
+        IEnumerable<TaskVlopper.Base.Model.Worklog> GetAllWorklogForGivenProjectAndUser(int projectId, string userId);
+        IEnumerable<TaskVlopper.Base.Model.Worklog> GetAllWorklogForGivenProjectAndTaskAndUser(int projectId, int taskId, string userId);
```
Comment (mszynka) [inline TaskVlopper.Base/Logic/IWorklogLogic.cs:13]: The same question why no overloaded method and 2 complicated names instead?

**Comment by mszynka** [comment]
Approved

File: TaskVlopper.Base/Logic/IMeetingLogic.cs (hunk 1 above)
Comment (bojakowsky) [inline TaskVlopper.Base/Logic/IMeetingLogic.cs:14]: cause it's not intuitive, after I'll implement them u'll see what i'm talking about

Comment (mszynka) [inline TaskVlopper.Base/Logic/IMeetingLogic.cs:14]: It's so intuitive that now we don't have Project.GetAll :smiley: So we need new interface in :cl: ProjectController

**Pull request merged**